contrib/upgrade.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+# this is an upgrade script, for gitea deployed on linux as systemd service
+# depends on: curl, xz, sha256sum, gpg
+# usage:      upgrade.sh [version]
+
+# change the variables below for your local setup
+giteaversion=${1:-1.15.10}
+giteabin=/usr/local/bin/gitea
+giteahome=/var/lib/gitea
+giteaconf=/etc/gitea/app.ini
+giteauser="git"
+giteacmd="sudo -u $giteauser $giteabin -c $giteaconf -w $giteahome"
+
+# download new binary
+binname=gitea-${giteaversion}-linux-amd64
+binurl="https://dl.gitea.io/gitea/${giteaversion}/${binname}.xz"
+echo downloading $binurl
+cd $giteahome # needed for gitea dump later
+curl -sSfL "$binurl" > ${binname}.xz
+curl -sSfL "${binurl}.sha256" > ${binname}.xz.sha256
+curl -sSfL "${binurl}.asc" > ${binname}.xz.asc
+
+# validate checksum & gpg signature (exit script if error)
+sha256sum -c ${binname}.xz.sha256
+gpg --keyserver keys.openpgp.org --recv 7C9E68152594688862D62AF62D9AE806EC1592E2
```

Comment:
Kind of a concern. Not sure if we should do a gpg import keys in a shell script. Also, `7C9E68152594688862D62AF62D9AE806EC1592E2`'s keys has a expiration(which is btw not a good thing for signing keys)  so in the case they don't push a non-expiring key to the keyservers and create a new key, this script will fail.

Reply:
Educate me if I'm wrong; my understanding is that importing a key just makes the key available locally for validation. It doesn't set a trust level on the key or anything. What is your concern?
I suspect that users who don't check this script to validate the key (by crosschecking with the docs - currently there is no better non-human source of truth afaik), wouldn't have checked the signature properly anyway. And if they trust the docs, they can equally trust this script, both live in the same repo.
About expiration: there is nothing I can change here; this is the key binaries are signed with. I'll add a `TODO` comment noting the expiration date

Reply:
So, yes on it's own importing a GPG key isn't a concern. However, certain systems are keen on importing GPG as you still can verify files and keys with. As well certain configurations will have a custom keyserver. I think it's best to mention it.

As for the expiration yeah, @go-gitea/owners will a new key be generated once this one is  expired?

Reply:
Ah ok, I get it now; people are afraid that having a key imported can make `gpg --verify` succeed on unrelated files.
We could drop the key after use again, not sure about how to proceed here :thinking:

Reply:
> Ah ok, I get it now; people are afraid that having a key imported can make `gpg --verify` succeed on unrelated files. We could drop the key after use again, not sure about how to proceed here thinking

I think it just should be mentioned and then let it be as-is.